File: python/arrays-and-hashing/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = defaultdict(lambda: 0)
        count[0] = 1
        prefix, answer = 0, 0
        for num in nums:
            prefix += num
            answer += count[prefix-k]
            count[prefix] += 1
        return answer

File: c++/arrays-and-hashing/subarray-sum-equals-k.cc
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        std::unordered_map<int, int> count = {{ 0, 1 }};
        int number = 0;
        int prefix = 0;

        for (int num : nums) {
            prefix += num;
            number += count[prefix - k];
            count[prefix] += 1;
        }

        return number;
    }
};
